### PR TITLE
[GPU] Fix some conv and pooling unit tests for DG2

### PR DESCRIPTION
**Pull request by tuxedcat**
### Details:
 - Adaptive tolerance in convolution test.
 - Change activation function into onednn supported one.

### Tickets:
 - 67491

### CODE REVIEW

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp [hunk 1]
```diff
@@ -477,7 +478,7 @@ TEST_P(conv_fp32_scale, basic) {
         reorder("reorder_bfyx", "scale", p.default_format, data_types::f32)
     );
 
-    tolerance = 1e-5f;
+    tolerance = p.default_type==data_types::f16 ? 1e-3f : 1e-5f;
```
Comment (sshlyapn) [inline src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp:481]: I think it may make sense to add common function for tolerance value, something like `get_default_tolerance(cldnn::data_types)`

Comment (tuxedcat) [reply src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp]: Thank you for your great suggestion. Done.

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp [hunk 2]
```diff
@@ -2947,19 +2949,19 @@ TEST_P(conv_int8_eltwise_onednn, u8_eltwise_prod_out) {
...
+    convolution_test_params{ CASE_CONV_S8S8_1, 3, 4 },
+    convolution_test_params{ CASE_CONV_S8S8_2, 3, 4 },
+    convolution_test_params{ CASE_CONV_S8S8_3, 3, 4 },
+
+    convolution_test_params{ CASE_CONV_U8S8_11, 3, 4 },
+    convolution_test_params{ CASE_CONV_U8S8_12, 3, 4 },
+    convolution_test_params{ CASE_CONV_U8S8_13, 3, 4 },
+    convolution_test_params{ CASE_CONV_S8S8_12, 3, 4 },
+    convolution_test_params{ CASE_CONV_S8S8_13, 3, 4 },
+    convolution_test_params{ CASE_CONV_S8S8_14, 3, 4 },
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp:2964]: Do we need to increase this value even if binary_mul fusion is enabled now?

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp [hunk 3]
```diff
@@ -3102,7 +3104,7 @@ TEST_P(conv_int8_quantize_u8_onednn, per_channel) {
         reorder("reorder_bfyx", "quantize", p.default_format, data_types::f32)
     );
 
-    tolerance = 1.0f;
+    tolerance = 1.f;
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp:3107]: why did you change this?

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp [hunk 4]
```diff
@@ -3075,7 +3077,7 @@ TEST_P(conv_fp32_activation_exp_onednn, basic) {
         reorder("reorder_bfyx", "activation", p.default_format, data_types::f32)
     );
 
-    tolerance = 1e-2f;
+    tolerance = default_tolerance(p.default_type);
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp:3080]: why did you change this? It is decreased from 1e-2 to 1e-5

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp (hunk 3 above)
Comment (tuxedcat) [reply src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp:3107]: Some code uses 1.f and the others uses 1.0f therefore I changed 1.0f to 1.f for code style consistency.

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp (hunk 4 above)
Comment (tuxedcat) [reply src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp:3080]: Testcases below are changed from 1e-2 to default_tolerance(p.default_type) and those tests are all passed.

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp (hunk 2 above)
Comment (tuxedcat) [reply src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp:2964]: binary_mul issue is for other testcase(conv_fp32_prelu_eltwise.basic_prod/0). In this test, changed value 2 to 3 because It seems that the best fusion senario of `input|conv|eltwise|crop|(reorder)` is `input|conv+eltwise|crop|(reorder)`.

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp (hunk 4 above)
Comment (tuxedcat) [reply src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp:3080]: conv_fp32_activation_abs_onednn
conv_fp32_activation_mish_onednn
conv_fp32_activation_swish_onednn
conv_fp32_activation_hswish_onednn
conv_fp32_activation_exp_onednn

**Review by isanghao — APPROVED** [review]
